include/directx/d3dx12.h

```diff
@@ -4037,6 +4037,665 @@ class CD3DX12_NODE_MASK_SUBOBJECT
+            // Must wait until session type count initialized
+            QueryProtectedResourceSessionTypes(i, m_dProtectedResourceSessionTypeCount[i].Count);
+        }
+
+        // Initialize features that requires highest version check
+        if (FAILED(m_hStatus = QueryHighestShaderModel())) {
+            return m_hStatus;
+        }
+
+        if (FAILED(m_hStatus = QueryHighestRootSignatureVersion())) {
+            return m_hStatus;
+        }
+
+        // Initialize Feature Levels data
+        if (FAILED(m_hStatus = QueryHighestFeatureLevel())) {
+            return m_hStatus;
+        }
+        
+        return m_hStatus;
+    }
+
+    HRESULT GetStatus() const
+    {
+        return m_hStatus;
+    }
+
+#ifdef FEATURE_SUPPORT_VARIANT_OPT_IN
+    static std::variant<CD3DX12FeatureSupport, HRESULT> Create(ID3D12Device* pDevice)
+    {
+        CD3DX12FeatureSupport features;
+        HRESULT result = features.Init(pDevice);
+        if (FAILED(result)) {
+            return result;
+        }
+
+        return features;
+    }
+#else
+    static CD3DX12FeatureSupport Create(ID3D12Device* pDevice)
+    {
+        CD3DX12FeatureSupport features;
+        features.Init(pDevice);
+        return features;
+    }
+#endif
+
+    // 0: D3D12_OPTIONS
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, DoublePrecisionFloatShaderOps);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, OutputMergerLogicOp);
+    FEATURE_SUPPORT_GET(D3D12_SHADER_MIN_PRECISION_SUPPORT, m_dOptions, MinPrecisionSupport);
+    FEATURE_SUPPORT_GET(D3D12_TILED_RESOURCES_TIER, m_dOptions, TiledResourcesTier);
+    FEATURE_SUPPORT_GET(D3D12_RESOURCE_BINDING_TIER, m_dOptions, ResourceBindingTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, PSSpecifiedStencilRefSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, TypedUAVLoadAdditionalFormats);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, ROVsSupported);
+    FEATURE_SUPPORT_GET(D3D12_CONSERVATIVE_RASTERIZATION_TIER, m_dOptions, ConservativeRasterizationTier);
+    FEATURE_SUPPORT_GET(UINT, m_dOptions, MaxGPUVirtualAddressBitsPerResource);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, StandardSwizzle64KBSupported);
+    FEATURE_SUPPORT_GET(D3D12_CROSS_NODE_SHARING_TIER, m_dOptions, CrossNodeSharingTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, CrossAdapterRowMajorTextureSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, VPAndRTArrayIndexFromAnyShaderFeedingRasterizerSupportedWithoutGSEmulation);
+    FEATURE_SUPPORT_GET(D3D12_RESOURCE_HEAP_TIER, m_dOptions, ResourceHeapTier);
+
+    // 1: Architecture
+    // Combined with Architecture1
+
+    // 2: Feature Levels
+    // Simply returns the highest supported feature level
+    D3D_FEATURE_LEVEL HighestFeatureLevel() const
+    {
+        return m_eMaxFeatureLevel;
+    }
+
+    // 3: Feature Format Support
+    HRESULT FormatSupport(DXGI_FORMAT Format, D3D12_FORMAT_SUPPORT1& Support1, D3D12_FORMAT_SUPPORT2& Support2)

```

Comment:
Continuation from the previous PR: FormatSupport, MultisampleQualityLevels, FormatInfo all use DXGI_FORMAT. Is it possible to combine them into one feature check?

Another concern is that there are two output parameters from this function (Support1 and Support2). Is that too complicated for the users?